### PR TITLE
Enhanced tile debug

### PR DESCRIPTION
**Pull request by amazingsmash**
More informative and accurate debug mesh on tiles, including grid, elevation data level, and tile level information.

### CODE REVIEW

**Comment by DiegoGomezDeck** [comment]
Q1:
- what about onGPUProgramHasChanged?  This "hack" is only needed for floatbuffers? What about other buffers and/or textures?

**Comment by DiegoGomezDeck** [comment]
Q2:
- Why, in java generated code, the AbstractGeometryMesh constructor is overloaded?  It's because of default value for arguments?  In this case, I don't like to have default-values in a protected constructor of an abstract class, those default values should be on the concrete subclasses.

File: iOS/G3MiOSSDK/Commons/Mesh/AbstractMesh.hpp [hunk 1]
```diff
@@ -46,7 +50,10 @@ class AbstractMesh : public Mesh {
                const IFloatBuffer* colors,
                const float colorsIntensity,
                bool depthTest,
-               const IFloatBuffer* normals);
+               const IFloatBuffer* normals,
+               bool polygonOffsetFill = false,
+               float polygonOffsetFactor = 0,
+               float polygonOffsetUnits = 0);
```
Comment (DiegoGomezDeck) [inline iOS/G3MiOSSDK/Commons/Mesh/AbstractMesh.hpp:57]: Please don't used default values in this constructor.  They should be on the constructors of the concrete subclasses.

**Comment by amazingsmash** [comment]
AbstractGeometryMesh fixed.

I've left a warning in useProgram() indicating that the OpenGL state that is shader specific, should be examined to determine which variables need a reset.

**Pull request merged**